Akkadu/rsi-api-widgets
# feat: set up rsi-api-interpreter.vue

## Description
* [notion page](https://www.notion.so/akkadu/RSI-API-Interpreter-Player-Logic-Front-End-0143c99869724b6d836da00010bcc033)
* I set up the directory structure
```
_rsi-api-widget

__modules

____rsi-api-base

____rsi-interpretation-manager

____rsi-interpretation-player

____rsi-api-react

____rsi-api-vue

__vue-example

__react-example
```
* I made the `interpretation player` widget front end
  * The Agora logic for that widget will be added in a future in another  PR.
 
* When the class `@akkadu/rsi-interpretation-player` is called it looks for the **dom ID** `akkadu-interpretation-player` and push the vanilla javascript player widget in the div.
* I made two packages: `rsi-api-vue` and `rsi-api-react`.
  * Theses two packages will be the one used by the users.

## How to test
* Go on the https://rsi-akkadu-vue-demo.netlify.app/#/ and https://rsi-akkadu-react-demo.netlify.app/#/
 * you should see the `interpretation player`,
 * when selecting a new language you should see the events in the console.

* You can check the documentation [here](https://rsi-akkadu-documentation.netlify.app/)  
* The diagram of the flow [here](https://whimsical.com/stream-rsi-api-consumer-interpreter-workflow-2uLJ3zrDn6WBWTYVtiTBXW) 


## Screenshoot

<img width="1140" alt="Screen Shot 2021-05-20 at 1 19 43 PM" src="https://user-images.githubusercontent.com/57079606/118923490-331cd200-b96e-11eb-95f2-149feb501602.png">
<img width="1020" alt="Screen Shot 2021-05-20 at 1 19 59 PM" src="https://user-images.githubusercontent.com/57079606/118923517-3c0da380-b96e-11eb-897b-b513d1d684a1.png">
